src/test/java/org/assertj/core/api/SoftAssertionsTest.java

```diff
@@ -1687,9 +1713,15 @@ public void map_soft_assertions_should_report_errors_on_final_methods_and_method
           // convert to Object otherwise will use extracting(String) in AbstractObjectAssert
           .extracting((Object) "a")
           .isEqualTo("456");
+    softly.assertThat(map)
+          .as("extracting(\"a\") as string")
+          .overridingErrorMessage("error message")
+          // convert to Object otherwise will use extracting(String, InstanceOfAssertFactory) in AbstractObjectAssert
+          .extracting((Object) "a", as(STRING))
```

Comment:
Thinking out loud, I wonder if we should rename `extracting` to `extractingByKey` to avoid the collision with `extracting(String, InstanceOfAssertFactory)` in `AbstractObjectAssert`, we would the same for `extracting(Object key)`, it would have the advantage to be obvious, still it might be confusing to see `extracting(Object )` from `AbstractObjectAssert` along with `extractingByKey`

We could even use `extractingByKey(K key)` since if you are extracting a key is should be compatible with the map key type.

If we go this way we would create another PR for renaming `extracting(Object key)` to `extractingByKey(K key)` but keeping the `extracting(Object key)` and deprecate it.

Reply:
You're right, the current `extracting(Object key)` is confusing also because it has an `extracting(String)` example in its javadoc 😅 (I just realized it)
I like `extractingByKey(K key)` also because of key type safety.

Shall I pause this PR and introduce `extractingByKey(K key)` first? Afterwards I can rebase this one adding the `InstanceOfAssertFactory` variant only to the new method.

Reply:
It does not matter if we do `extractingByKey(K, InstanceOfAssertFactory)` first and then `extractingByKey(K)` as long as we do both for 3.14.

Goo point for renaming `extracting(Object... keys)` too.

Let's finish this PR first, see if there is no issue using the `K` generic type instead of `Object` and then move on to `extractingByKey(K)`.

I'm planning to release 3.14 soon as soon as we have these changes done.

Reply:
About `extracting(Object... keys)`, it should become `extractingByKeys(Object... keys)`, right?

Reply:
should be `extractingByKeys(K... keys)`, I suspect will lead to a javac warning about heap pollution as in `extracting(Function...)`

Reply:
Yes I meant with the generic 😅 ok will do!

About 3.14 release, would you like to include also the remaining in #1605 for `IterableAssert` and `OptionalAssert`or shall these go into the next one?

Reply:
I think for consistency it would be better to finish #1605, how long do you think it's going to take ? (don't feel pressure, just want to have a gut feeling about it)

Reply:
I would say additional 7-10 days, not because of complexity but due to the limited time I will have in the upcoming week

Reply:
I tried to close this PR first but having partial renaming was a bit confusing, so I preferred to do the deprecations first in #1623.

Reply:
PR ready again! #1623 should be checked and merged first, then I will probably need to rebase this one.

Reply:
yup, I have just commented it